tools/system_libs.py

```diff
@@ -1729,11 +1730,14 @@ def retrieve():
 
     def check_tag():
       if is_tarbz2:
-        names = tarfile.open(fullpath, 'r:bz2').getnames()
+        with ToolchainProfiler.profile_block('check_tag_tar_bz2(' + name + ')'):
+          names = tarfile.open(fullpath, 'r:bz2').getnames()
       elif url.endswith('.tar.gz'):
-        names = tarfile.open(fullpath, 'r:gz').getnames()
+        with ToolchainProfiler.profile_block('check_tag_tar_gz(' + name + ')'):
+          names = tarfile.open(fullpath, 'r:gz').getnames()
       else:
-        names = zipfile.ZipFile(fullpath, 'r').namelist()
+        with ToolchainProfiler.profile_block('check_tag_zip(' + name + ')'):
+          names = zipfile.ZipFile(fullpath, 'r').namelist()
```

Comment:
This is just be one single block instead?  (the exact filename should fairly obvious in "name" anywa).

Reply:
It could, but it would not be as informative, I wanted to highlight the different archive names there, for some reason zips tend to run really fast, whereas .tar.bz2 is really slow.

Reply:
OK.

Hopefully we can completely remove this code anyway.   It seems crazy open these zip files for each emcc call .   We souldn't even need to keep the zipfiles around.

I can take a look at patch to completely remove `check_tag` .. unless you are already doing this.